pkg/storage/etcd3/watcher.go

```diff
@@ -180,6 +180,9 @@ func (wc *watchChan) startWatching() {
 			wc.sendEvent(parseEvent(e))
 		}
 	}
+	wc.cancel()
```

Comment:
Yes. That's the assumption.
Let me think about how to better refactor the code.